File: Day_1/Go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	currentPosition := 50
	numZeroes := 0
	numZeroes2 := 0
	const MOD = 100

	file, err := os.Open("./input.txt")
	if err != nil {
		fmt.Printf("Error opening file: %v\n", err)
		return
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := strings.TrimSpace(scanner.Text())

		if line == "" {
			continue
		}

		direction := line[0]
		distanceStr := line[1:]
		distance, err := strconv.Atoi(distanceStr)
		if err != nil {
			continue
		}

		switch direction {
		case 'R':
			numZeroes2 += (currentPosition + distance) / MOD
			currentPosition = (currentPosition + distance) % MOD

		case 'L':
			numZeroes2 += (distance / 100)
			rest := distance % 100

			if currentPosition > 0 && rest >= currentPosition {
				numZeroes2++
			}

			currentPosition = ((currentPosition-distance)%MOD + MOD) % MOD
		}

		if currentPosition == 0 {
			numZeroes++
		}

		fmt.Printf("Line: %s | Count: %d \n", line, currentPosition)
	}

	if err := scanner.Err(); err != nil {
		fmt.Printf("Error reading line: %v\n", err)
		return
	}

	fmt.Printf("Part 1 : Final Password (Total Zeroes): %d\n", numZeroes)
	fmt.Printf("Part 2 : Final Password (Total Zeroes): %d\n", numZeroes2)
}

File: Day_1/python/main.py
def get_answer_1():
    count_zeroes = 0
    current_pos = 50
    MOD = 100

    with open("./input.txt") as f:
        lines = [line.strip() for line in f if line.strip()]

    for line in lines:
        direction = line[0]
        distance = int(line[1:])

        if direction == "L":
            current_pos = (current_pos - distance) % MOD
        elif direction == "R":
            current_pos = (current_pos + distance) % MOD
        if current_pos == 0:
            count_zeroes += 1

    return count_zeroes


def get_answer_2():
    count_zeroes = 0
    current_pos = 50
    MOD = 100

    with open("./input.txt") as f:
        lines = [line.strip() for line in f if line.strip()]

    for line in lines:
        direction = line[0]
        distance = int(line[1:])

        if direction == "L":
            count_zeroes += distance // 100
            rest = distance % 100

            if current_pos > 0 and rest >= current_pos:
                count_zeroes += 1

            current_pos = (current_pos - distance) % MOD
        elif direction == "R":
            count_zeroes += (current_pos + distance) // 100
            current_pos = (current_pos + distance) % MOD

    return count_zeroes


print(get_answer_1())
print(get_answer_2())
